awx/main/tasks.py

```diff
@@ -2221,3 +2222,10 @@ def build_env(self, instance, **kwargs):
 
     def build_cwd(self, instance, **kwargs):
         return settings.BASE_DIR
+
+
+celery_app.register_task(RunJob())
```

Comment:
I know that one of the side effects of this task registry is persistence in the form of re-using instances of the task (and thus not calling `__init__`) again.

Is there any risk to our jobs when persisting state between invocations of these tasks?

Reply:
@matburt: I believe celery already did this when inheriting from `Task`, initializing the instance only once. For an invocation of the task, we pass everything via arguments between the methods, and don't store anything using `self` for this reason.